Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp

```diff
@@ -1951,31 +1951,33 @@ void MediaPlayerPrivateGStreamer::handleMessage(GstMessage* message)
         asyncStateChangeDone();
         break;
     case GST_MESSAGE_STATE_CHANGED: {
-        if (!messageSourceIsPlaybin || m_isDelayingLoad)
-            break;
-        updateStates();
-
-        // Construct a filename for the graphviz dot file output.
         GstState newState;
         gst_message_parse_state_changed(message, &currentState, &newState, nullptr);
-        CString dotFileName = makeString(GST_OBJECT_NAME(m_pipeline.get()), '.',
-            gst_element_state_get_name(currentState), '_', gst_element_state_get_name(newState)).utf8();
-        GST_DEBUG_BIN_TO_DOT_FILE_WITH_TS(GST_BIN(m_pipeline.get()), GST_DEBUG_GRAPH_SHOW_ALL, dotFileName.data());
-
-        if (!m_isLegacyPlaybin && currentState == GST_STATE_PAUSED && newState == GST_STATE_PLAYING)
-            playbin3SendSelectStreamsIfAppropriate();
 
 #if USE(GSTREAMER_HOLEPUNCH)
         if (currentState == GST_STATE_NULL && newState == GST_STATE_READY) {
             // If we didn't create a video sink, store a reference to the created one.
             if (!m_videoSink)
-                g_object_get(m_pipeline.get(), "video-sink", &m_videoSink.outPtr(), nullptr);
+                g_object_get(pipeline(), "video-sink", &m_videoSink.outPtr(), nullptr);
 
             // Ensure that there's a buffer with the transparent rectangle available when playback is going to start.
-            pushNextHolePunchBuffer();
+            if (m_videoSink)
+                pushNextHolePunchBuffer();
         }
 #endif
 
+        if (!messageSourceIsPlaybin || m_isDelayingLoad)
```

Comment:
Have into account that if you move this condition here, the holepunch code block is going to run when every single element of the pipeline changes its state. Are you sure that's what you want? Wouldn't you prefer to compare the name of the element that changed state against some pattern and only run the holepunch block for those selected elements?

[Reading createHolePunchVideoSink()](https://github.com/WebPlatformForEmbedded/WPEWebKit/blob/wpe-2.28/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp#L3516), it looks that this holepunch codeblock is mostly relevant on Nexus, which is the only case where createHolePunchVideoSink() returns null and the sink is expected to be found in this code block.

@magomez, you're more acquainted with the holepunch code than me. What do you think?